src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1591,7 +1603,9 @@ private async ValueTask<(Stream, TransportContext?)> ConnectAsync(HttpRequestMes
                 stream = sslStream;
             }
 
-            return (stream, transportContext);
+            static IPEndPoint? GetRemoteEndPoint(Stream stream) => (stream as NetworkStream)?.Socket?.RemoteEndPoint as IPEndPoint;
```

Comment:
`EstablishSocksTunnel` should be easy to light up here as well.

Should we try to extract the info from `EstablishProxyTunnelAsync` as well?

Reply:
> `EstablishSocksTunnel` should be easy to light up here as well.

Pushed an untested change in 8a3fa5fca34bc5db485a54011d3338994b9b00c7. Can you recommend/push a test?

> Should we try to extract the info from `EstablishProxyTunnelAsync` as well?

It should be logged logged as part of the inner CONNECT request. Feels complicated to modify all `SendAsync` call chains so they can return the endpoint for this scenario to the outer request, I don't think it's worth it.

Reply:
> Feels complicated to modify all SendAsync call chains so they can return the endpoint

I was thinking we could try to extract it from the stream itself since these are all internals only.

```c#
static IPEndPoint? GetRemoteEndPoint(Stream? stream)
{
    if (stream is HttpContentStream contentStream)
    {
        stream = contentStream._connection?._stream;
    }

    return (stream as NetworkStream)?.Socket?.RemoteEndPoint as IPEndPoint;
}
```